tests/check_check_tags.c

```diff
@@ -0,0 +1,315 @@
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+#include "../lib/libcompat.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <check.h>
+#include <assert.h>
+#include "check_check.h"
+
+/* from check_check_master.c */
+#define MAXSTR 300
+char * escape_percent(const char *original, size_t original_size);
+
+
+START_TEST(red_test1)
+{
+  ck_assert_msg(0, "Red fail");
+}
+END_TEST
+
+START_TEST(blue_test1)
+{
+  ck_assert_msg(0, "Blue fail");
+}
+END_TEST
+
+START_TEST(purple_test1)
+{
+  ck_assert_msg(0, "Purple fail");
+}
+END_TEST
+
+START_TEST(yellow_test1)
+{
+  ck_assert_msg(0, "Yellow fail");
+}
+END_TEST
+
+START_TEST(black_test1)
+{
+  ck_assert_msg(0, "Black fail");
+}
+END_TEST
+
+
+static Suite *make_tagged_suite(void)
+{
+  Suite *s;
+
+  TCase *red, *blue, *purple, *yellow, *black;
+
+  s = suite_create("Check Tag Filtering");
+
+  red = tcase_create_tagged("Red", "Red");
+  suite_add_tcase (s, red);
+  tcase_add_test(red, red_test1);
+
+  blue = tcase_create_tagged("Blue", "Blue");
+  suite_add_tcase (s, blue);
+  tcase_add_test(blue, blue_test1);
+
+  purple = tcase_create_tagged("Purple", "Red Blue");
+  suite_add_tcase (s, purple);
+  tcase_add_test(purple, purple_test1);
+
+  yellow = tcase_create_tagged("Yellow", "Yellow");
+  suite_add_tcase (s, yellow);
+  tcase_add_test(yellow, yellow_test1);
+
+  black = tcase_create("Black");
+  suite_add_tcase (s, black);
+  tcase_add_test(black, black_test1);
+
+  return s;
+}
+
+static   SRunner *sr = NULL;
+
+static void tag_test_setup(void)
+{
+  Suite *s;
+
+  s = make_tagged_suite();;
+  sr = srunner_create(s);
+  srunner_set_fork_status(sr, CK_NOFORK);
+
+}
+
+static void tag_test_teardown (void)
+{
+  srunner_free (sr);
+}
+
+/*
+ * Show that with no filter we run all the tests
+ */
+START_TEST(no_filter)
+{
+  const char *expected_msgs[5] = { "Red fail",
+				   "Blue fail",
+				   "Purple fail",
+				   "Yellow fail",
+				   "Black fail"};
+  TestResult **tr_fail_array;
+  unsigned int ntests_failed;
+  unsigned int ntests_run;
+  size_t n_expected_msgs;
+  const char *got_msg;
+  unsigned int i;
+  TestResult *tr;
+
+
+  n_expected_msgs = sizeof(expected_msgs)/sizeof(expected_msgs[0]);
+
+  srunner_run(sr, NULL, NULL, NULL, NULL, CK_VERBOSE);
```

Comment:
I would like to see a test where an empty include tag is passed, or tags which do not exist are passed, and thus no tests are run. There should also be tests for the exclusion tags as well. 

Also, what happens if one separates tags with multiple spaces? Can there be a test for this as well?

Also, what if someone passes "Yellow" as both an included and excluded test? What should happen? Is this defined? Can there be a test for this as well?